### PR TITLE
expression: fix VecEvalString PadCharToFullLength

### PR DESCRIPTION
**Pull request by b41sh**
### What problem does this PR solve? 
In `PAD_CHAR_TO_FULL_LENGTH` sql mod, VecEvalString return string without pad to length.
This will affect the outcome of builtin Comapre String functions in PR [12875](https://github.com/pingcap/tidb/pull/12875)

### Check List 

Tests 

 - Unit test
 - Integration test

### CODE REVIEW

**Comment by qw4990** [comment]
PTAL @XuHuaiyu

File: expression/column_test.go [hunk 1]
```diff
@@ -214,3 +214,26 @@ func (s *testEvaluatorSuite) TestColHybird(c *C) {
...
+	}
+	result, err := newBuffer(types.ETString, 1024)
+	c.Assert(err, IsNil)
+	c.Assert(col.VecEvalString(ctx, input, result), IsNil)
+
+	it := chunk.NewIterator4Chunk(input)
+	for row, i := it.Begin(), 0; row != it.End(); row, i = it.Next(), i+1 {
+		v, _, err := col.EvalString(ctx, row)
+		c.Assert(err, IsNil)
+		c.Assert(v, Equals, result.GetString(i))
```
Comment (XuHuaiyu) [inline expression/column_test.go]: Should we add a line like `c.Assert(len(v), Equals, xx)` before this line.

Comment (b41sh) [reply expression/column_test.go]: ok

**Pull request merged**